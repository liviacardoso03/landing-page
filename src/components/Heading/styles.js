import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,

  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,

  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,

  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
  //'se uppercase for verdadeiro, o texto vai parar uppercase, se for falso, nenhuma alteração é feita'
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    //'se o tema for colorDark, ele vai ser primaryColor, se ele não for white, ele será white'

    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
