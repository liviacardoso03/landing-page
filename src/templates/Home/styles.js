import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  //Funções para poder utilizar JavaScript aqui dentro
  //Foi criada uma função para que fosse possível ter acesso as props

  ${({ theme, background }) => css`
    background: ${theme.colors.secondaryBg};
  `}
`;
