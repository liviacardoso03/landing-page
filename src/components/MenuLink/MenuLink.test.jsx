import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { MenuLink } from '.';

describe('<MenuLink />', () => {
  it('should render a link', () => {
    //'deve renderizar um link'

    // renderTheme(<MenuLink link="https://www.google.com.br">Children</MenuLink>);
    // expect(screen.getByRole('link', { name: 'Children' })).toBeInTheDocument();
    //'espero que o link esteja no documento na tela'

    renderTheme(<MenuLink link="https://www.google.com.br">Children</MenuLink>);
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute('target', '_self');
    //'espero que o link tenha o seguinte atributo: target e _self'
  });

  it('should render open in a new tab', () => {
    //'deve renderizar ao abrir uma nova aba'

    renderTheme(
      <MenuLink link="https://www.google.com.br" newTab={true}>
        Children
      </MenuLink>,
    );
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute('target', '_blank');
    //'espero que o link tenha o seguinte atributo: target e _blank'
  });

  it('should match with snapshot', () => {
    const { container } = renderTheme(
      <MenuLink link="http://localhost" newTab={false}>
        Children
      </MenuLink>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
