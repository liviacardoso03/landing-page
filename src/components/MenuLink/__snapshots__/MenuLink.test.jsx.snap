// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match with snapshot 1`] = ` `;
